Assets/Src/Scripts/AgileMvvm/ViewModel.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace AgileMvvm {
+  // The ViewModel class that maintains models and raises model update events to the bound Views. It
+  // also provides the utility methods to manage the binding relationships.
+  //
+  // In AgileMvvm, the relationships among View, ViewModel and Model are:
+  //
+  // * The Model does not know about the View or the ViewModel.
+  // * The View does not know about the Model, but knows the ViewModel.
```

Comment:
It's not necessary that View knows the ViewModel. They can be bound by the third parities, and don't know each others.

Reply:
Yes, the term "know" is a bit confusing here.

The MVVM pattern has different definitions and implementations. In some implementations, the View can really own a reference of the ViewModel. In other cases, the View just communicates with the ViewModel via one-way or two-way bindings. Of course, the latter one is recommended in modern architecture.

In both cases, it's still able to say that "the View knows the ViewModel" since there could be two-way communications. But since "know" is confusing here, I rewrote and simplified the comments. Please take another look.